app/controllers/catalog_controller.rb

```diff
@@ -379,7 +379,7 @@ def st_edit
   def st_form_field_changed
     return unless load_edit("st_edit__#{params[:id]}", "replace_cell__explorer")
     @group_idx = false
-    default_entry_point("generic") if params[:display]
+    default_entry_point("generic", "composite") if params[:display]
```

Comment:
The only time `prov_type`would be `nil` is for service bundles which is what this code path is handling by passing `generic`.  

The default entry point values should always be set, I do not see any reason to wait until the display flag has been checked.  Is that something you are planning to fix in this PR?